blenderproc/python/loader/HavenMaterialLoader.py

```diff
@@ -21,65 +20,71 @@ def load_haven_mat(folder_path: str = "resources/haven", used_assets: list = [],
     # makes the integration of complex materials easier
     addon_utils.enable("node_wrangler")
 
-    folder_path = resolve_path(folder_path)
+    textures_folder_path = resolve_path(folder_path)
 
     if preload and fill_used_empty_materials:
         raise Exception("Preload and fill used empty materials can not be done at the same time, check config!")
-    if os.path.exists(folder_path) and os.path.isdir(folder_path):
-        for asset in os.listdir(folder_path):
-            if used_assets:
-                skip_this_one = True
-                for used_asset in used_assets:
-                    if asset.startswith(used_asset):
-                        skip_this_one = False
-                        break
-                if skip_this_one:
-                    continue
-            current_path = os.path.join(folder_path, asset)
-            if os.path.isdir(current_path):
-                # find the current base_image_path by search for _diff_, this make it independent of the used res
-                all_paths = glob.glob(os.path.join(current_path, "*.jpg"))
-                base_image_path = ""
-                for path in all_paths:
-                    if "_diff_" in path:
-                        base_image_path = path
-                        break
-                if not os.path.exists(base_image_path):
-                    continue
-
-                # if the material was already created it only has to be searched
-                if fill_used_empty_materials:
-                    new_mat = MaterialLoaderUtility.find_cc_material_by_name(asset, add_cp)
-                else:
-                    new_mat = MaterialLoaderUtility.create_new_cc_material(asset, add_cp)
-                if preload:
-                    # if preload then the material is only created but not filled
-                    continue
-                elif fill_used_empty_materials and not MaterialLoaderUtility.is_material_used(new_mat):
-                    # now only the materials, which have been used should be filled
-                    continue
-
-                # construct all image paths
-                # the images path contain the words named in this list, but some of them are differently
-                # capitalized, e.g. Nor, NOR, NoR, ...
```

Comment:
Could we add this docu back, it is still valid as far as I can tell.

Reply:
Added this document back at line 77 in the new `identify_texture_maps()` function.